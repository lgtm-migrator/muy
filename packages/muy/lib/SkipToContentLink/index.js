import React from "react"
import PropTypes from "prop-types"
import Link from "@material-ui/core/Link"

const SkipToContentLink = ({ children, href, ...props }) => (
  <Link href={href} variant={"srOnly"} {...props}>
    {children}
  </Link>
)

export default SkipToContentLink

SkipToContentLink.defaultProps = {
  children: <>Skip to content</>,
  href: "#content",
}

SkipToContentLink.propTypes = {
  children: PropTypes.node.isRequired,
  href: PropTypes.string,
}
