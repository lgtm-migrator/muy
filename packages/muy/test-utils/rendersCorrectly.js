import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
