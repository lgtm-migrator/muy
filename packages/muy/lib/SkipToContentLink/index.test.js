import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import SkipToContentLink from "."

describe("<SkipToContentLink />", () => {
  it("should render correctly with no props", () => {
    rendersCorrectly(<SkipToContentLink />)
  })
  it("should render correctly with href prop", () => {
    rendersCorrectly(<SkipToContentLink href={"#content"} />)
  })
  it("should render correctly with different href prop", () => {
    rendersCorrectly(<SkipToContentLink href={"#another-anchor-id"} />)
  })
  it("should render correctly with children", () => {
    rendersCorrectly(<SkipToContentLink>Skip to content</SkipToContentLink>)
  })
  it("should render correctly with different children", () => {
    rendersCorrectly(
      <SkipToContentLink>Another skip link text</SkipToContentLink>
    )
  })
})
