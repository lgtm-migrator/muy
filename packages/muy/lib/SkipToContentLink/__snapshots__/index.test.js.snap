// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkipToContentLink /> should render correctly with children 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-srOnly MuiTypography-colorPrimary"
  href="#content"
  onBlur={[Function]}
  onFocus={[Function]}
>
  Skip to content
</a>
`;

exports[`<SkipToContentLink /> should render correctly with different children 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-srOnly MuiTypography-colorPrimary"
  href="#content"
  onBlur={[Function]}
  onFocus={[Function]}
>
  Another skip link text
</a>
`;

exports[`<SkipToContentLink /> should render correctly with different href prop 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-srOnly MuiTypography-colorPrimary"
  href="#another-anchor-id"
  onBlur={[Function]}
  onFocus={[Function]}
>
  Skip to content
</a>
`;

exports[`<SkipToContentLink /> should render correctly with href prop 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-srOnly MuiTypography-colorPrimary"
  href="#content"
  onBlur={[Function]}
  onFocus={[Function]}
>
  Skip to content
</a>
`;

exports[`<SkipToContentLink /> should render correctly with no props 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-srOnly MuiTypography-colorPrimary"
  href="#content"
  onBlur={[Function]}
  onFocus={[Function]}
>
  Skip to content
</a>
`;
